tests/unit/package_managers/test_generic.py

```diff
@@ -44,10 +46,44 @@
     version: '1.0'
 artifacts:
     - download_url: https://example.com/artifact
+      target: archive.zip
       checksums:
         md5: 3a18656e1cea70504b905836dee14db0
 """
 
+LOCKFILE_INVALID_TARGET = """
+metadata:
+    version: '1.0'
+artifacts:
+    - download_url: https://example.com/artifact
+      target: ./../../../archive.zip
+      checksums:
+        md5: 3a18656e1cea70504b905836dee14db0
+"""
+LOCKFILE_TARGET_OVERLAP = """
+metadata:
+    version: '1.0'
+artifacts:
+    - download_url: https://example.com/artifact
+      target: archive.zip
+      checksums:
+        md5: 3a18656e1cea70504b905836dee14db0
+    - download_url: https://example.com/artifact2
+      target: archive.zip
+      checksums:
+        md5: 3a18656e1cea70504b905836dee14db0
+"""
+
+LOCKFILE_WRONG_CHECKSUM = """
+metadata:
+    version: '1.0'
+artifacts:
+    - download_url: https://example.com/artifact
+      target: archive.zip
+      checksums:
+        md5: 32112bed1914cfe3799600f962750b1d
+"""
+
```

Comment:
I'd also add one with a complex download URL to test path extraction that is to serve as the target file name.

Reply:
Added to the `LOCKFILE_VALID` case